tests/ctst/world/Zenko.ts

```
@@ -99,6 +106,318 @@ export default class Zenko extends World {
+            CacheHelper.ARWWI[ARWWIName] = this.parameters.AssumedSession;
+            this.cliMode.parameters.AssumedSession = CacheHelper.ARWWI[ARWWIName];
+            this.cliMode.assumed = true;
+        } else {
+            this.parameters.AssumedSession = CacheHelper.ARWWI[ARWWIName];
+            this.cliMode.parameters.AssumedSession = CacheHelper.ARWWI[ARWWIName];
+            this.cliMode.assumed = true;
+        }
+    }
+
+    /**
+     * HTTP client to request JWT token given the username and password.
+     *
+     * @param {string} username - username of user requesting token
+     * @param {string} password - password of user requesting token
+     * @param {string} host - host URL of keycloak service
+     * @param {number} port - port of keycloak service
+     * @param {string} path - path of keycloak service authentication API
+     * @param {string} clientId - id of the client of the user
+     * @param {string} grantType - grant of the user
+     * @returns {string} the OIDC token
+     */
+    async getWebIdentityToken(
+        username: string,
+        password: string,
+        host: string,
+        port: string,
+        path: string,
+        clientId: string,
+        grantType: string,
+    ) {
+        const baseUrl = this.parameters.ssl === false ? 'http://' : 'https://';
+        const data = qs.stringify({
+            username,
+            password,
+            // eslint-disable-next-line camelcase
+            client_id: clientId,
+            // eslint-disable-next-line camelcase
+            grant_type: grantType,
+        });
+        const config: AxiosRequestConfig = {
+            method: 'post',
+            url: `${baseUrl}${host}:${port}${path}`,
+            headers: {
+                'Content-Type': 'application/x-www-form-urlencoded',
+            },
+            data,
+        };
+        const result: AxiosResponse = await axios(config);
+        return result.data.access_token;
```

Do we not care about any error handling here, considering we are making an API request for returning an access_token and we possibly have many functions relying on it?

PS: Not reviewed how it is used yet.

---

Usually the approach for tests, from my experience, is "let it throw", as it will return an error anyway...

It may be better to throw a known and human-readable error though, to ease the debugging, as long as enough information is logged along with the message

---

yeah! aiming for easy debugging as well.
If you think error is clear enough, we should be good.
Feel free to resolve the comment if that's the case.

---

455f9416a39b4307accf57abb26b7b87a86b3662

thoughts?

---

perfect!